crm_phonecall_planner/wizards/crm_phonecall_planner.py

```diff
@@ -0,0 +1,212 @@
+                ("partner_id", "!=", False),
+            ],
+            order="date",
+        )
+        # Get partners to plan
+        partner_domain = safe_eval(self.res_partner_domain) + [
+            ("phonecall_calendar_ids", "!=", False),
+        ]
+        forbidden_partners = existing_calls.mapped("partner_id")
+        partners = Partner.search(partner_domain)
+        # And now the hot chili...
+        while partners and now <= end:
+            now += call_duration
+            _logger.debug(
+                "Plannig phonecalls for %s",
+                fields.Datetime.to_string(now),
+            )
+            # Should we continue tomorrow?
+            if not start.time() <= now.time() <= end.time():
+                _logger.info(
+                    "Finished plannig phonecalls for %s with %d calls so far",
+                    fields.Date.to_string(now.date()),
+                    len(self.planned_calls),
+                )
+                now = datetime.combine(now.date(), start.time()) + tomorrow
+                continue
+            # Know partners that we cannot call right now
+            if self.repeat_calls:
+                forbidden_partners = Phonecall.search(utm_domain + [
+                    ("partner_id", "in", partners.ids),
+                    ("date", ">",
+                     fields.Datetime.to_string(now - repetition_gap)),
+                ]).mapped("partner_id")
+            # Know partners we can call right now
+            available_partners = Partner.with_context(now=now).search([
+                ("id", "in", partners.ids),
+                ("id", "not in", forbidden_partners.ids),
+                ("phonecall_available", "=", True),
+            ])
+            # Continue when nobody is available
+            if not available_partners:
+                continue
+            # Just pick the first one and continue
+            if not self.repeat_calls:
+                winner = available_partners[:1]
+                self._schedule_call(winner, now)
+                partners -= winner
+                continue
+            # Get a partner with no calls, or with the oldest one
+            self.env.cr.execute(
+                oldest_call_to_partner.format(
+                    ",".join(["%s"] * len(available_partners)),
+                ),
+                available_partners.ids,
+            )
+            winner = Partner.browse(self.env.cr.fetchone()[0])
+            self._schedule_call(winner, now)
+        _logger.info("Total planned phonecalls: %d", len(self.planned_calls))
+        return {
+            "name": "Generated calls",
+            "type": "ir.actions.act_window",
+            "res_model": "crm.phonecall",
+            "views": [[False, "tree"], [False, "calendar"], [False, "form"]],
+            "domain": [("id", "in", self.planned_calls.ids)],
+        }
+
+    @api.multi
+    def _schedule_call(self, partner, when):
+        call = self.env["crm.phonecall"]
```

Comment:
To reduce cyclomatic complexity, do:
```python
if not partner:
    return self.env["crm.phonecall"]
_logger.debug(
...
```

Reply:
I think I'll just remove the returned value, it's not used after all.